src/plugins/intel_gna/src/serial/gna_model_serial.cpp

```diff
@@ -563,61 +609,49 @@ void GNAModelSerial::Export(const GnaAllocations& allocations, std::ostream& os)
+        node.scale_factor = ep.scaleFactor;
+        node.model_precision =
             InferenceEngine::Precision(static_cast<InferenceEngine::Precision::ePrecision>(ep.precision));
-        input.set_precision(ep.element_size);
-        input.model_layout = static_cast<InferenceEngine::Layout>(ep.layout);
-        input.allocated_size = input.get_required_size();
+        node.set_precision(ep.element_size);
+        node.model_layout = static_cast<InferenceEngine::Layout>(ep.layout);
+        node.allocated_size = node.get_required_size();
 
         auto inputDims = InferenceEngine::SizeVector();
         for (auto i = 0; i < ep.shape.NumberOfDimensions; ++i) {
             inputDims.push_back(ep.shape.Dimensions[i]);
         }
-        input.dims = inputDims;
+        node.dims = inputDims;
 
         // read tensor names
         for (uint8_t tId = 0; tId < ep.tensor_names_count; ++tId) {
-            input.tensor_names.insert(readString(is));
-        }
-
-        AppendTensorNameIfNeeded(input);
-    }
-}
-
-void GNAModelSerial::ImportOutputs(std::istream& is, void* basePtr, GnaOutputs& outputs) {
-    for (auto& output : outputs.Get()) {
-        header_latest::RuntimeEndPoint ep = ReadEndPoint(is);
-
-        output.ptrs.push_back(reinterpret_cast<float*>(reinterpret_cast<uint8_t*>(basePtr) + ep.descriptor_offset));
-        output.orientation = ep.orientation;
-        output.num_elements = ep.elements_count;
-        output.scale_factor = ep.scaleFactor;
-        output.set_precision(ep.element_size);
-        output.model_precision =
-            InferenceEngine::Precision(static_cast<InferenceEngine::Precision::ePrecision>(ep.precision));
-        output.model_layout = static_cast<InferenceEngine::Layout>(ep.layout);
-        output.allocated_size = output.get_required_size();
-
-        auto outputDims = InferenceEngine::SizeVector();
-        for (auto i = 0; i < ep.shape.NumberOfDimensions; ++i) {
-            outputDims.push_back(ep.shape.Dimensions[i]);
+            node.tensor_names.insert(readString(is));
         }
-        output.dims = outputDims;
-
-        // read tensor names
-        for (uint8_t tId = 0; tId < ep.tensor_names_count; ++tId) {
-            output.tensor_names.insert(readString(is));
+        AppendTensorNameIfNeeded(node);
+
+        // read preprocessing model
+        if (model_header_.version.major == 2 && model_header_.version.minor >= 9) {
+            std::string ir_xml_str = readString(is);
+            if (!ir_xml_str.empty()) {
+                // read IR bin
+                size_t ir_bin_size = 0;
+                readBits(ir_bin_size, is);
+
+                ov::Tensor ir_bin_tensor(ov::element::u8, ov::Shape({ir_bin_size}));
+                readNBytes(ir_bin_tensor.data(), ir_bin_size, is);
+
+                // restore model
+                ov::Core core;
+                node.pre_post_process_model = core.read_model(ir_xml_str, ir_bin_tensor);
```

Comment:
1. I just wonder if it is not too expensive to create Core for each input separately.